modules/Bio/EnsEMBL/Compara/RunnableDB/HomologyAnnotation/MakeDiamondDBPerGenomeDB.pm

```diff
@@ -0,0 +1,74 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+=pod
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::MakeDiamondDBPerGenomeDB
+
+=head1 DESCRIPTION
+
+Runnable wrapper for DumpMembersIntoFasta per genome_db and generate DIAMOND database
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::MakeDiamondDBPerGenomeDB;
+
+use warnings;
+use strict;
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+use Data::Dumper;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::DumpMembersIntoFasta');
+
+sub run {
+    my $self = shift;
+    $self->SUPER::run;
+
+    my $fasta_file    = $self->param('fasta_file');
+    my $diamond_exe   = $self->param('diamond_exe');
+    my $query_db_dir  = $self->param('query_db_dir');
+    my $genome_db_id  = $self->param('genome_db_id');
```

Comment:
Based on the base runnable, the parameter can be either `genome_db_id` or `genome_db_ids`. If you want only to accept this one in this case, then I would do `param_required`.